src/listeners/handlers/local-file-upload-handler.ts

```diff
@@ -115,6 +95,44 @@ class LocalFileUploadHandler implements IHandler {
         const splitPath: string[] = submissionUri.pathname.split("/");
         return splitPath[splitPath.length - 1];
     }
+
+    handle(msg: string): Promise<boolean> {
+        return LocalFileUploadHandler._parseAmqpMessage(msg)
+            .then(plan => this.processUploadJobsSequential(plan.jobs));
+    }
+
+    processUploadJobsSequential(jobs: Job[]): Promise<true> {
+        if (jobs.length == 0) {
+            return Promise.resolve(true);
+        } else {
+            const job: Job = R.head(jobs)!;
+            return this.doLocalFileUpload(job)
+                .then(() => {
+                    return this.processUploadJobsSequential(R.tail(jobs))
+                })
+                .catch(error => {
+                    console.error("An error occured: ", error);
+                    return Promise.reject(error);
+                })
+        }
+    }
+
+    doLocalFileUpload(job: Job): Promise<boolean> {
+        return this._maybeDownloadFiles(job, this.dirBasePath)
```

Comment:
Since this is using the strategy of checking if the files/conversions exist locally, rather than downloading each time, we need considerations for disk-space and delete/clean-up.